src/python/pants/util/dirutil.py

```diff
@@ -500,3 +500,10 @@ def split_basename_and_dirname(path):
   if not os.path.isfile(path):
     raise ValueError("{} does not exist or is not a regular file.".format(path))
   return (os.path.dirname(path), os.path.basename(path))
+
+
+def narrow_relative_paths(cur_root_dir, new_root_subdir, rel_paths):
```

Comment:
This is pretty specific... would not add it here unless you've used the pattern multiple times.

Reply:
Reverted in 97787f5!